### PR TITLE
Refactors the Palette comopnent in ColorPicker to remove the tailwind dependency.

### LINKED ISSUE #2390: Refactor the Palette component in ColorPicker to remove the Tailwind dependency for the colors.
@praveen-murali-ind as per our discussion let's refactor the Palette component to avoid the tailwind dependency and implement a more straightforward and simple way similar to what the `react-colorful` has implemented for the color boxes.

This includes the change in the color list prop of the palette to an array of color values and using those instead of the color bg classes and gradient as an inline style. These are breaking changes, we will have to refactor the usages in NeetoDesk and NeetoWireframe.

cc: @josephmathew900

_Originally posted by @deepakjosp in https://github.com/bigbinary/neeto-ui/issues/2385#issuecomment-2503437030_

### PR DESCRIPTION
**Pull request by deepakjosp**
Fixes #2390 

**Description**
- Updates the color palette to support hex values and removes the tailwind dependency.

**Checklist**

- [x] I have made corresponding changes to the documentation.
- [x] I have updated the types definition of modified exports.
- [x] I have verified the functionality in some of the neeto web-apps.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [ ] ~I have added proper `data-cy` and `data-testid` attributes.~
- [x] I have added the necessary label (`patch`/`minor`/`major` - If package publish
      is required).

**Reviewers**

### CODE REVIEW

**Comment by deepakjosp** [comment]
@josephmathew900 _a please review. Please don't merge. I'll merge it once the product PRs are approved.

File: src/components/ColorPicker/Palette.jsx
Comment (josephmathew900) [inline src/components/ColorPicker/Palette.jsx:19]: Will this work?

```suggestion
      const isActive = Boolean(
        (color?.hex && color.hex === hex) ||
          (color?.colorClassName && color.colorClassName === colorClassName)
      );
```

**Review by josephmathew900 — APPROVED** [review]
@deepakjosp _a Added a minor suggestion. Rest LGTM.

**Pull request merged**
